### PR TITLE
[Woo POS] M2: Implement payment success screen as per design

### LINKED ISSUE #12220: [Woo POS] Do not use hardcoded color for "success" screen
`WooPosPaymentSuccessScreen`

We have to declare this color in the theme with darker color for the dark mode

### LINKED ISSUE #12222: [Woo POS] M2: Implement Payment success screen as per design
https://www.figma.com/design/TfaZ4LUkEwEGrxfnEFzvJj/WooCommerce-POS?node-id=2877-89385&t=B6Q2oh658RIviTet-0

Image

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #12222
Closes: #12220 

### Description

The PR implements the UI and animation (in a simplified form, as we decided to not prioritize this for now) of the Success payment screen

TfaZ4LUkEwEGrxfnEFzvJj-fi-2877_89385

### Testing information

* Collect payment
* Notice the updated success screen with it's animation. Validate that the UI follows the design

### Images/gif

https://github.com/user-attachments/assets/48cfa08e-9730-4294-9957-9a2d00d9a101

- [x] I have considered adding unit tests for this change. If I decided not to add them, I have provided a brief explanation below (optional):
- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/229">19.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeProductInfoDialog.kt [hunk 1]
```diff
@@ -214,7 +214,7 @@ fun ProductInfoDialogPreview() {
             modifier = Modifier.fillMaxSize(),
             contentAlignment = Alignment.Center
         ) {
-            ProductInfoDialog(
+            WooPosProductInfoDialog(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeProductInfoDialog.kt:217]: @AnirudhBhat jfyi I fixed the naming to follow the convention

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt [hunk 2]
```diff
@@ -1,112 +1,205 @@
...
+    val size by animateDpAsState(
+        targetValue = if (animationStarted) 156.dp else 0.dp,
+        label = "Check mark size"
+    )
 
-        OutlinedButton(
-            modifier = Modifier
-                .padding(24.dp.toAdaptivePadding())
-                .height(80.dp.toAdaptivePadding())
-                .width(600.dp.toAdaptivePadding()),
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt:89]: @backwardstruck Setting adaptive paddings to the "sizes" may not be a good idea, especially on the buttons, as those must have specific physical sizes to be easy-clickable. We may want to set it only on the padding/margins

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt [hunk 3]
```diff
@@ -1,112 +1,205 @@
...
+}
 
-        Spacer(modifier = Modifier.height(24.dp.toAdaptivePadding()))
+@Composable
+private fun CheckMarkIcon(
+    animationStarted: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    val size by animateDpAsState(
+        targetValue = if (animationStarted) 156.dp else 0.dp,
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt]: ❓ Looks like the Check mark icon container size is 165 DP in Figma.

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov. LGTM. I've just left one non-blocking comment.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt (hunk 3 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt]: Good catch! Changed it to 164, as 165 can not be in 4dp grid, so probably a small mistake in the design

08747f0dbbdff1d4d53dbc4faf6f9c6366934886

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt (hunk 2 above)
Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/payment/success/WooPosTotalsPaymentSuccessScreen.kt:89]: Good point! I'll double check what I did here.

**Pull request merged**